### ISSUE TITLE
Auto-completion does not work with coc

### ISSUE THREAD

**Issue by rene-atzenhofer** [wsdjeg/SpaceVim#2991]
## Expected behavior, english is recommend
The auto-completion list should popup after I started typing the first few characters.

When I the log file from coc with CocOpenLog I get the following error. Coc works fine without SpaceVim.
```
2019-08-07T12:58:43.754 ERROR (pid:10631) [workspace] - Error on create buffer: TypeError: this._additionalKeywords is not iterable
    at Document.setIskeyword (/home/user/.cache/vimfiles/.cache/init.vim/.dein/build/index.js:49901:29)
    at Document.init (/home/user/.cache/vimfiles/.cache/init.vim/.dein/build/index.js:49893:14)
    at processTicksAndRejections (internal/process/task_queues.js:86:5)
```

## Environment Information

- OS: linux
- vim version: -
- neovim version: 0.4.0
- SpaceVim version: 1.3.0-dev
- SpaceVim status: 

```
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
```

## The reproduce ways from Vim starting (Required!)
1. Set "coc" as autocomplete_method
2. Open a ts file in vim
3. Start typing in Insert

When I try the steps above with Deoplete it works.

## Output of the `:SPDebugInfo!`

<details><summary> SpaceVim debug information </summary>

### SpaceVim options :

```toml
  auto_disable_touchpad = 1
  autocomplete_method = 'deoplete'
  autocomplete_parens = 1
  buffer_index_type = 4
  checkinstall = 1
  colorscheme = 'gruvbox'
  colorscheme_bg = 'dark'
  colorscheme_default = 'desert'
  commandline_prompt = '➭'
  custom_color_palette = []
  custom_plugins = []
  debug_level = 1
  default_indent = 2
  disabled_plugins = []
  enable_ale = 0
  enable_bepo_layout = 0
  enable_cursorcolumn = 0
  enable_cursorline = 1
  enable_debug = 0
  enable_googlesuggest = 0
  enable_guicolors = 0
  enable_insert_leader = 1
  enable_javacomplete2_py = 0
  enable_key_frequency = 0
  enable_language_specific_leader = 1
  enable_neocomplcache = 0
  enable_neomake = 1
  enable_os_fileformat_icon = 0
  enable_powerline_fonts = 1
  enable_statusline_bfpath = 0
  enable_statusline_mode = 0
  enable_statusline_tag = 1
  enable_tabline_filetype_icon = 0
  enable_tabline_ft_icon = 0
  enable_vimfiler_filetypeicon = 1
  enable_vimfiler_gitstatus = 1
  enable_vimfiler_welcome = 0
  enable_ycm = 0
  error_symbol = '✖'
  expand_tab = 1
  filemanager = 'vimfiler'
  filetree_direction = 'right'
  filetype_icons = {}
  force_global_config = 0
  gitcommit_issue_icon = ''
  gitcommit_pr_icon = ''
  github_username = ''
  guifont = ''
  hiddenfileinfo = 1
  home_files_number = 6
  hosts_url = 'https://userw.githubusercontent.com/usercaljk/hosts/master/hosts'
  info_symbol = 'ⓘ'
  keep_server_alive = 1
  language = ''
  leader_guide_default_group_name = ''
  leader_guide_flatten = 1
  leader_guide_hspace = 5
  leader_guide_max_size = 0
  leader_guide_position = 'botright'
  leader_guide_run_map_on_popup = 1
  leader_guide_sort_horizontal = 0
  leader_guide_submode_mappings = {'<C-C>': 'win_close'}
  leader_guide_vertical = 0
  lint_on_save = 1
  lint_on_the_fly = 0
  max_column = 120
  plugin_bundle_dir = '/home/user/.cache/vimfiles/'
  plugin_manager = 'dein'
  plugin_manager_processes = 16
  plugin_name = 'open-browser.vim'
  project_rooter_automatically = 1
  project_rooter_patterns = ['.git/', '_darcs/', '.hg/', '.bzr/', '.svn/', '.SpaceVim.d/']
  realtime_leader_guide = 1
  relativenumber = 1
  search_tools = ['rg', 'ag', 'pt', 'ack', 'grep', 'findstr']
  sidebar_direction = ''
  sidebar_width = 30
  simple_mode = 0
  smartcloseignoreft = ['tagbar', 'vimfiler', 'defx', 'SpaceVimRunner', 'SpaceVimREPL', 'SpaceVimQuickFix', 'HelpDescribe', 'VebuggerShell', 'VebuggerTerminal', 'SpaceVimTabsManager']
  smartcloseignorewin = ['__Tagbar__', 'vimfiler:default']
  snippet_engine = 'neosnippet'
  src_root = 'E:\sources\'
  statusline_inactive_separator = 'bar'
  statusline_iseparator = 'nil'
  statusline_left_sections = ['winnr', 'filename', 'major mode', 'syntax checking', 'minor mode lighters']
  statusline_right_sections = ['fileformat', 'cursorpos', 'percentage']
  statusline_separator = 'nil'
  statusline_unicode_symbols = 0
  terminal_cursor_shape = 2
  update_retry_cnt = 3
  version = '1.3.0-dev'
  vim_help_language = 'en'
  vimcompatible = 0
  warning_symbol = '⚠'
  wildignore = '*/tmp/*,*.so,*.swp,*.zip,*.class,tags,*.jpg,*.ttf,*.TTF,*.png,*/target/*,.git,.svn,.hg,.DS_Store,*.svg'
  windows_index_type = 3
  windows_leader = 's'
  windows_smartclose = 'q'
```

### SpaceVim layers :

```toml
[[layers]]
  name="autocomplete"
[[layers]]
  name="checkers"
[[layers]]
  name="format"
[[layers]]
  name="edit"
[[layers]]
  name="ui"
[[layers]]
  name="core"
[[layers]]
  name="core#banner"
[[layers]]
  name="core#statusline"
[[layers]]
  name="core#tabline"
```

### SpaceVim Health checking :

SpaceVim clipboard support check report:
Checking +clipboard:
      SUCCEED!
SpaceVim environment check report:
Current progpath: nvim(/usr/bin/nvim)
version: 800
OS: linux
[shell, shellcmdflag, shellslash]: ['/bin/zsh', '-c', 0]
SpaceVim lua support check report:
Checking +lua:
      Failed : Known issue, neovim do not support lua now.
SpaceVim python support check report:
Checking +python3:
      SUCCEED!
Checking +python:
      SUCCEED!

### SpaceVim runtime log :

```log
[ SpaceVim ] : logger file  does not exists, only log for current process will be shown!
[ SpaceVim ] [12:56:42] [ Info ] Startup with no argv, current dir is used: ~/
[ SpaceVim ] [12:56:42] [ Info ] find local conf: /home/user/SpaceVim.d/init.toml
[ SpaceVim ] [12:56:42] [ Info ] generate local conf: /home/user/.cache/SpaceVim/conf/home_user_SpaceVim_d_init_toml.json
[ SpaceVim ] [12:56:42] [ Info ] start to apply config [local]
[ SpaceVim ] [12:56:42] [ Info ] SpaceVim server startup at:/tmp/spacevim_nvim_server
[ SpaceVim ] [12:56:42] [ Info ] Start to find root for: 
[ SpaceVim ] [12:56:42] [ Info ] change to root:/home/user
[ SpaceVim ] [12:56:42] [ Info ] try to open SpaceVim welcome page
[ SpaceVim ] [12:56:42] [ Info ] change to root:/home/user
[ SpaceVim ] [12:56:44] [ Info ] Start to find root for: /home/user/Working/greeter.ts
[ SpaceVim ] [12:56:44] [ Info ] change to root:/home/user
[ SpaceVim ] [12:57:45] [ Info ] Start to find root for: 
[ SpaceVim ] [12:57:45] [ Info ] change to root:/home/user
[ SpaceVim ] [12:57:49] [ Info ] change to root:/home/user
[ SpaceVim ] [12:57:51] [ Info ] Start to find root for: /tmp/nvimNbTb0r/2/issue_report.md
[ SpaceVim ] [12:57:51] [ Info ] change to root:/home/user
[ SpaceVim ] [12:57:51] [ Info ] Start to find root for: /tmp/nvimNbTb0r/2/issue_report.md
[ SpaceVim ] [12:57:51] [ Info ] change to root:/home/user
[ SpaceVim ] [12:58:31] [ Info ] Start to find root for: /home/user/SpaceVim.d/init.toml
[ SpaceVim ] [12:58:31] [ Info ]         (.git/):/.git/
[ SpaceVim ] [12:58:31] [ Info ] change to root:
[ SpaceVim ] [12:58:38] [ Info ] change to root:/home/user
[ SpaceVim ] [12:58:40] [ Info ] change to root:
[ SpaceVim ] [12:58:43] [ Info ] Start to find root for: /tmp/nvimNbTb0r/4/issue_report.md
[ SpaceVim ] [12:58:43] [ Info ] change to root:
```
</details>

**Comment by gugahoa** [comment]
Seems like it's a coc.nvim issue: https://github.com/neoclide/coc.nvim/issues/1050

**Comment by ghedamat** [comment]
I can confirm this issue as well, but no errors for me the completion simply does nothing

**Comment by rene-atzenhofer** [comment]
> Seems like it's a coc.nvim issue: [neoclide/coc.nvim#1050](https://github.com/neoclide/coc.nvim/issues/1050)

Thank you. I also found this issue and verified that I am on the latest version, so this shouldn't be the reason for the error. Today I tested it again and now it works without any issues. So, unfortunately I cannot say what caused the error, but I am happy that it works now.


> I can confirm this issue as well, but no errors for me the completion simply does nothing

I am not sure whether your issue is related to this error or not. Today I just updated the plugins with :SPUpdate and the completion works now with coc.
